teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt

```diff
@@ -0,0 +1,216 @@
+                }
+            }
+
+            override fun onFailure(exception: Exception) {
+            }
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_ar_map)
+
+        back.setOnClickListener {
+            onBackPressed()
+        }
+        mapView.onCreate(savedInstanceState)
+        start_ar.setOnClickListener {
+            if (currentRoute == null) {
+                Toast.makeText(this, "Route is not ready yet!", Toast.LENGTH_LONG).show()
+            } else {
+                ArNavigationActivity.directionsRoute = currentRoute
+                ArNavigationActivity.start(this)
+            }
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        mapView.onStart()
+        mapView.getMapAsync(this)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        mapView.onResume()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        mapView.onPause()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        mapView.onStop()
+        arLocationEngine.removeLocationUpdates(locationCallback)
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        mapView.onDestroy()
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        mapView.onSaveInstanceState(outState)
+    }
+
+    override fun onBackPressed() {
+        super.onBackPressed()
+        overridePendingTransition(0, 0)
+    }
+
+    override fun onMapClick(destination: LatLng): Boolean {
+        destinationMarker?.let(mapboxMap::removeMarker)
+        destinationMarker = mapboxMap.addMarker(MarkerOptions().position(destination))
+
+        if (originPoint == null) {
+            Toast.makeText(this, "Source location is not determined yet!", Toast.LENGTH_LONG).show()
+            return false
+        }
+
+        getRoute(
+            origin = originPoint!!,
+            destination = Point.fromLngLat(destination.longitude, destination.latitude)
+        )
+
+        start_ar.visibility = View.VISIBLE
+
+        return true
+    }
+
+    override fun onMapReady(mapboxMap: MapboxMap) {
+        this.mapboxMap = mapboxMap
+        mapboxMap.setStyle(Style.Builder().fromUri(MAP_STYLE)) {
+            enableLocationComponent()
+        }
+
+        mapboxMap.addOnMapClickListener(this)
+    }
+
+    private fun getRoute(origin: Point, destination: Point) {
+        NavigationRoute.builder(this)
+            .accessToken(Mapbox.getAccessToken()!!)
+            .origin(origin)
+            .destination(destination)
+            .build()
+            .getRoute(object : Callback<DirectionsResponse> {
+                override fun onResponse(call: Call<DirectionsResponse>, response: Response<DirectionsResponse>) {
+                    if (response.body() == null || response.body()!!.routes().size < 1) {
```

Comment:
```suggestion
                    if (response.body() == null || response.body()!!.routes().size == 0) {
```
nit: Maybe it's more obvious

Reply:
or just `response.body()!!.routes().isEmpty()`

Reply:
fixed